aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -0,0 +1,247 @@
+module AptosFramework::Stake {
+    use Std::Vector;
+    use Std::Signer;
+    use CoreFramework::SystemAddresses;
+    use CoreFramework::Timestamp;
+//    use CoreFramework::Reconfiguration;
+    use AptosFramework::TestCoin::{Self, Coin};
+
+    const MINIMUM_LOCK_PERIOD: u64 = 86400;
+    const MINIMUM_RECONFIG_PERIOD: u64 = 3600;
+
+    /// Basic unit of stake delegation, it's stored in StakeInfo.
+    struct Delegation has store {
+        coin: Coin,
+        from: address,
+        locked_until_secs: u64,
+    }
+
+    /// Aggregation of delegation and represent a validator's voting power, stored in ValidatorInfo.
+    /// Invariants:
+    /// 1. current_stake = sum(active + cooldown)
+    /// 2. user interact with warmup and inactive if it's in the ValidatorSet.
+    /// 3. user interact with active, inactive if it's not in the ValidatorSet.
+    /// 4. warmup and cooldown are empty if it's not in the ValidatorSet.
+    struct StakeInfo has store {
+        // sum of active and cooldown stakes, updated on epoch boundary.
+        current_stake: u64,
+        // active stake
+        active: vector<Delegation>,
+        // inactive stake, can be withdrawn
+        inactive: vector<Delegation>,
+        // pending activation for next epoch
+        warmup: vector<Delegation>,
+        // pending deactivation for next epoch
+        cooldown: vector<Delegation>,
+    }
+
+    /// Consensus information per validator, stored in validator address.
+    struct ValidatorInfo has key {
+        stake_info: StakeInfo,
+        consensus_pubkey: vector<u8>,
+        network_address: vector<u8>,
+    }
+
+    /// Full ValidatorSet, stored in @CoreResource.
+    struct ValidatorSet has key {
+        consensus_scheme: u8,
+        // minimum stakes requires to join validator set
+        minimum_stake: u64,
+        // maximum stakes allowed to join validator set
+        maximum_stake: u64,
+        // should we store the ValidatorInfo in place?
+        validators: vector<address>,
```

Comment:
this gets tricky, since you want to gather all the consensus keys, but if we support too many validators it might be a better operation to leave out of the module and at the storage read layer

I had a proposal back in the day that we'd have a proposed validator info and current validator info in each validators account but we narrowed in on just copying it to the active validator set

Reply:
having the address only makes it more scalable right? then we query storage again for the key/network address